test/t8_forest/t8_gtest_element_is_leaf.cxx

```diff
@@ -0,0 +1,151 @@
+    if (t8_cmesh_is_empty (cmesh)) {
+      /* forest_commit does not support empty cmeshes, we skip this case */
+      GTEST_SKIP ();
+    }
+    /* Build the default scheme (TODO: Test this with all schemes) */
+    scheme = t8_scheme_new_default_cxx ();
+    forest = t8_forest_new_uniform (cmesh, scheme, level, 0, sc_MPI_COMM_WORLD);
+    t8_forest_ref (forest);
+    //const int maxlevel = t8_forest_get_maxlevel (forest);
+    int maxlevel = 7;
+    const int recursive_adapt = 1;
+    forest_adapt = t8_forest_new_adapt (forest, t8_test_adapt_first_child, recursive_adapt, 0, &maxlevel);
+  }
+
+  void
+  TearDown () override
+  {
+    if (t8_cmesh_is_empty (cmesh)) {
+      t8_cmesh_destroy (&cmesh);
+    }
+    else {
+      t8_forest_unref (&forest);
+      t8_forest_unref (&forest_adapt);
+    }
+  }
+
+  t8_cmesh_t cmesh;
+  t8_forest_t forest, forest_adapt;
+  t8_scheme_cxx_t *scheme;
+};
+
+void
+t8_test_element_is_leaf_for_forest (t8_forest_t forest)
+{
+  const t8_locidx_t num_local_trees = t8_forest_get_num_local_trees (forest);
+
+  for (t8_locidx_t itree = 0; itree < num_local_trees; ++itree) {
+    const t8_locidx_t num_elements_in_tree = t8_forest_get_tree_num_elements (forest, itree);
+    const t8_eclass_t tree_class = t8_forest_get_tree_class (forest, itree);
+    const t8_eclass_scheme_c *scheme = t8_forest_get_eclass_scheme (forest, tree_class);
+    /* Allocate memory to build a non-leaf element. */
+    t8_element_t *not_leaf;
+    scheme->t8_element_new (1, &not_leaf);
+    /* Iterate over all the tree's leaf elements, check whether the leaf
+     * is correctly identified by t8_forest_element_is_leaf,
+     * build its parent and its first child (if they exist), and verify
+     * that t8_forest_element_is_leaf returns false. */
+    for (t8_locidx_t ielement = 0; ielement < num_elements_in_tree; ++ielement) {
+      const t8_element_t *leaf_element = t8_forest_get_element_in_tree (forest, itree, ielement);
+      EXPECT_TRUE (t8_forest_element_is_leaf (forest, leaf_element, itree));
+      /* Compute parent and first child of element and check that they are not in the tree */
+      const int element_level = scheme->t8_element_level (leaf_element);
+      if (element_level > 0) {
+        scheme->t8_element_parent (leaf_element, not_leaf);
+        EXPECT_FALSE (t8_forest_element_is_leaf (forest, not_leaf, itree));
+      }
+      if (element_level < scheme->t8_element_maxlevel ()) {
+        scheme->t8_element_child (leaf_element, 0, not_leaf);
+        EXPECT_FALSE (t8_forest_element_is_leaf (forest, not_leaf, itree));
+      }
+    }
+    scheme->t8_element_destroy (1, &not_leaf);
+  }
+}
+
+TEST_P (element_is_leaf, element_is_leaf)
+{
+  t8_test_element_is_leaf_for_forest (forest);
+}
+
+TEST_P (element_is_leaf, element_is_leaf_adapt)
+{
+  t8_test_element_is_leaf_for_forest (forest_adapt);
+}
+
+/* TODO: Figure out how to do pretty printing for the cmesh examples when they are in a combined parameter. */
```

Comment:
What is the status of this TODO, is it helpful here, or better in a location for all cmeshes?

Reply:
Something like this 
```suggestion
auto pretty_print_params = [] (const testing::TestParamInfo<std::tuple<int, cmesh_example_base *>> &info) {
  std::string name = std::string ("Level_") + std::to_string (std::get<0> (info.param));
  std::string cmesh_name;
  std::get<1> (info.param)->param_to_string (cmesh_name);
  name += std::string ("_") + cmesh_name;
  return name;
};
```
should solve it

Reply:
Thank you @Davknapp for the suggestion!